llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4632,10 +4644,9 @@ LoopVectorizationCostModel::computeMaxVF(ElementCount UserVF, unsigned UserIC) {
   // found modulo the vectorization factor is not zero, try to fold the tail
   // by masking.
   // FIXME: look for a smaller MaxVF that does divide TC rather than masking.
-  if (Legal->prepareToFoldTailByMasking()) {
-    CanFoldTailByMasking = true;
+  selectTailFoldinStyle();
```

Comment:
Can this be moved before computing the max VF, i.e. outside this function?

Reply:
No, tried to do it - causes lots of the spurious error messages. computeMaxVF contains some early exits in the code above, which affects this function results.